Networking/Networking/Mapper/AIProductMapper.swift

```diff
@@ -0,0 +1,223 @@
+    let categories: [String]
+
+    init(name: String,
+         description: String,
+         shortDescription: String,
+         virtual: Bool,
+         shipping: Shipping,
+         tags: [String],
+         price: String,
+         categories: [String]) {
+        self.name = name
+        self.description = description
+        self.shortDescription = shortDescription
+        self.virtual = virtual
+        self.shipping = shipping
+        self.tags = tags
+        self.price = price
+        self.categories = categories
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        let name = container.failsafeDecodeIfPresent(String.self, forKey: .name) ?? ""
+        let description = container.failsafeDecodeIfPresent(String.self, forKey: .description) ?? ""
+        let shortDescription = container.failsafeDecodeIfPresent(String.self, forKey: .shortDescription) ?? ""
+        let virtual = container.failsafeDecodeIfPresent(Bool.self, forKey: .virtual) ?? false
+        let shipping = try container.decode(Shipping.self, forKey: .shipping)
+        let tags = container.failsafeDecodeIfPresent([String].self, forKey: .tags) ?? []
+        let price = container.failsafeDecodeIfPresent(targetType: String.self,
+                                                      forKey: .price,
+                                                      alternativeTypes: [.decimal(transform: { NSDecimalNumber(decimal: $0).stringValue })]) ?? ""
+        let categories = container.failsafeDecodeIfPresent([String].self, forKey: .categories) ?? []
+
+        self.init(name: name,
+                  description: description,
+                  shortDescription: shortDescription,
+                  virtual: virtual,
+                  shipping: shipping,
+                  tags: tags,
+                  price: price,
+                  categories: categories)
+    }
+
+    enum CodingKeys: String, CodingKey {
+        case name = "name"
+        case description = "description"
+        case shortDescription = "short_description"
+        case virtual = "virtual"
+        case shipping = "shipping"
+        case tags = "tags"
+        case price = "price"
+        case categories = "categories"
+    }
+}
+
+// MARK: Helper to init Product
+//
+private extension Product {
+    init(siteID: Int64,
+         aiProduct: AIProduct,
+         categories: [ProductCategory],
+         tags: [ProductTag]) {
+        self.init(siteID: siteID,
+                  productID: 0,
+                  name: aiProduct.name,
+                  slug: "",
+                  permalink: "",
+                  date: Date(),
+                  dateCreated: Date(),
+                  dateModified: nil,
+                  dateOnSaleStart: nil,
+                  dateOnSaleEnd: nil,
+                  productTypeKey: ProductType.simple.rawValue,
+                  statusKey: ProductStatus.published.rawValue,
```

Comment:
👀 Do you think we should keep the status as draft so that we don't have to update it later when saving it as draft on the UI layer?

Reply:
I agree. Done in https://github.com/woocommerce/woocommerce-ios/pull/10770/commits/b651a6f817eee96c3d2e93066d743413106e9a84